src/tools/live-utils/start-session.ts

```diff
@@ -1,47 +1,251 @@
+// File: src/tools/live-utils/start-session.ts
 import { sanitizeUrlParam } from "../../lib/utils";
 import logger from "../../logger";
 import childProcess from "child_process";
+import { getLiveData } from "./device-cache";
+import { resolveVersion } from "./version-resolver";
+import { customFuzzySearch } from "../../lib/fuzzy";
 
-interface StartSessionArgs {
-  browser: string;
+export interface DesktopArgs {
+  platformType: "desktop";
+  url: string;
   os: string;
   osVersion: string;
-  url: string;
+  browser: string;
   browserVersion: string;
   isLocal: boolean;
 }
+export interface MobileArgs {
+  platformType: "mobile";
+  url: string;
+  os: string;
+  osVersion: string;
+  device: string;
+  browser: string;
+  isLocal: boolean;
+}
 
+/**
+ * Entrypoint: detects platformType & delegates.
+ */
 export async function startBrowserSession(
-  args: StartSessionArgs,
+  args: DesktopArgs | MobileArgs,
 ): Promise<string> {
-  // Sanitize all input parameters
-  const sanitizedArgs = {
-    browser: sanitizeUrlParam(args.browser),
-    os: sanitizeUrlParam(args.os),
-    osVersion: sanitizeUrlParam(args.osVersion),
+  if (args.platformType === "desktop") {
+    const entry = await filterDesktop(args);
+    const url = buildDesktopUrl(args, entry);
+    openBrowser(url);
+    if (entry.notes) {
+      return `${url}, ${entry.notes}`;
+    }
+    return url;
+  } else {
+    const entry = await filterMobile(args);
+    const url = buildMobileUrl(args, entry);
+    openBrowser(url);
+    if (entry.notes) {
+      return `${url}, ${entry.notes}`;
+    }
+    return url;
+  }
+}
+
+// ——— Desktop ———
+
+interface DesktopEntry {
+  os: string;
+  os_version: string;
+  browser: string;
+  browser_version: string;
+  notes?: string;
+}
+
+async function filterDesktop(args: DesktopArgs): Promise<DesktopEntry> {
```

Comment:
filtering code should not be part of this file. Create new one and also break it down, function is too long and there is too much logic here.